android/engine/src/main/java/org/smartregister/fhircore/engine/ui/pin/PinLoginScreen.kt

```diff
@@ -0,0 +1,270 @@
+const val PIN_SET_PIN_CONFIRM_BUTTON = "setPinConfirmButton"
+
+@Composable
+fun PinLoginScreen(viewModel: PinViewModel) {
+
+  val showError by viewModel.showError.observeAsState(initial = false)
+
+  PinLoginPage(
+    onPinChanged = viewModel::onPinChanged,
+    showError = showError,
+    onMenuLoginClicked = { viewModel.onMenuLoginClicked() },
+    forgotPin = viewModel::forgotPin,
+  )
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun PinLoginPage(
+  modifier: Modifier = Modifier,
+  onPinChanged: (String) -> Unit,
+  showError: Boolean = false,
+  onMenuLoginClicked: () -> Unit,
+  forgotPin: () -> Unit
+) {
+
+  var showMenu by remember { mutableStateOf(false) }
+  var showForgotPinDialog by remember { mutableStateOf(false) }
+
+  Surface(color = colorResource(id = R.color.white_slightly_opaque)) {
+    TopAppBar(
+      title = { Text(text = "", Modifier.testTag(PIN_TOOLBAR_TITLE)) },
+      navigationIcon = {
+        IconButton(onClick = {}) {
+          Icon(
+            Icons.Filled.ArrowBack,
+            contentDescription = "Back arrow",
+            modifier = Modifier.size(0.dp).testTag(PIN_TOOLBAR_MENU_ICON)
+          )
+        }
+      },
+      actions = {
+        IconButton(
+          onClick = { showMenu = !showMenu },
+          modifier = Modifier.testTag(PIN_TOOLBAR_MENU_BUTTON)
+        ) { Icon(imageVector = Icons.Outlined.MoreVert, contentDescription = null) }
+        DropdownMenu(
+          expanded = showMenu,
+          onDismissRequest = { showMenu = false },
+          Modifier.testTag(PIN_TOOLBAR_MENU)
+        ) {
+          DropdownMenuItem(
+            onClick = {
+              showMenu = false
+              onMenuLoginClicked()
+            },
+            modifier = Modifier.testTag(PIN_TOOLBAR_MENU_LOGIN)
+          ) { Text(text = stringResource(id = R.string.pin_menu_login)) }
+        }
+      }
+    )
+
+    if (showForgotPinDialog) {
+      ForgotPinDialog(forgotPin = forgotPin, onDismissDialog = { showForgotPinDialog = false })
+    }
+
+    Column(
+      modifier =
+        Modifier.fillMaxSize()
+          .padding(horizontal = 16.dp, vertical = 70.dp)
+          .wrapContentWidth(Alignment.CenterHorizontally)
+    ) {
+      Image(
+        painter = painterResource(id = R.drawable.ic_liberia),
+        contentDescription = stringResource(id = R.string.app_logo),
+        modifier =
+          modifier
+            .padding(top = 16.dp)
+            .align(Alignment.CenterHorizontally)
+            .requiredHeight(120.dp)
+            .requiredWidth(140.dp)
+            .testTag(APP_LOGO_TAG),
+      )
+      Text(
+        text = stringResource(R.string.eCBIS_app_name),
+        textAlign = TextAlign.Center,
+        fontWeight = FontWeight.Bold,
+        fontSize = 22.sp,
+        modifier = modifier.padding(top = 16.dp).align(Alignment.CenterHorizontally)
+      )
+
+      Text(
+        text = stringResource(R.string.enter_pin_w4vv01),
```

Comment:
Is it necessary to use hardcoded string "Enter pin for W4VV-01" here or it's going to be different for different users?
@dubdabasoduba

Reply:
CC: @f-odhiambo

Reply:
We can change the message to "Enter pin for ${logined_user}. I assume we can get this from the details stored on the shared preferences the `${logined_user} can be the user's username.  

If we can't get the `username` then we can have it say " Enter the users pin"

Reply:
I think these should be configurable and note hardcoded , but let @dubdabasoduba give final word on this